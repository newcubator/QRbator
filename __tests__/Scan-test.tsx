import { act, render } from "@testing-library/react-native";
import ScanScreen from "../app/scan";

jest.mock("expo-camera", () => ({
  Camera: {
    requestCameraPermissionsAsync: jest.fn(),
  },
  CameraView: jest.fn().mockImplementation(({ children }) => children),
}));

describe("<ScanScreen />", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    require("expo-camera").Camera.requestCameraPermissionsAsync.mockReset();
  });

  test("renders permission denied state", async () => {
    require("expo-camera").Camera.requestCameraPermissionsAsync.mockResolvedValue(
      {
        status: "denied",
      }
    );

    const renderResult = render(<ScanScreen />);

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0));
    });

    expect(renderResult.getByText("noCameraPermission")).toBeTruthy();
  });

  test("renders with camera when permission granted", async () => {
    require("expo-camera").Camera.requestCameraPermissionsAsync.mockResolvedValue(
      {
        status: "granted",
      }
    );

    const renderResult = render(<ScanScreen />);

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0));
    });

    expect(renderResult.getByText("positionQRCode")).toBeTruthy();
  });

  test("matches snapshot with permission denied", async () => {
    require("expo-camera").Camera.requestCameraPermissionsAsync.mockResolvedValue(
      {
        status: "denied",
      }
    );

    const component = render(<ScanScreen />);

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0));
    });

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
