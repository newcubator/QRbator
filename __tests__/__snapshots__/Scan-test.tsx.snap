// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScanScreen /> matches snapshot with permission denied 1`] = `
<RCTSafeAreaView
  className="flex-1 items-center justify-center bg-corp-white px-6"
>
  <Text
    className="mb-4 text-center text-lg text-corp-grey"
  >
    noCameraPermission
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      className="ml-2 font-semibold text-corp-white"
      style={
        [
          undefined,
        ]
      }
    >
      goBack
    </Text>
  </View>
</RCTSafeAreaView>
`;
